### DISCUSSION TITLE
Default Window Size too small

### DISCUSSION THREAD

**Discussion by Drallas** [wezterm/wezterm#2311] [General]
I would like to set the Default window size bigger, is this possible, can't find it in the docs?

**Comment by wez** [comment]
The easiest thing is:
* [initial_cols](https://wezfurlong.org/wezterm/config/lua/config/initial_cols.html)
* [initial_rows](https://wezfurlong.org/wezterm/config/lua/config/initial_rows.html)

More advanced thing is: (requires a nightly build):
* [gui-startup](https://wezfurlong.org/wezterm/config/lua/gui-events/gui-startup.html)

**Comment by Drallas** [comment]
Thanks I added:

initial_cols = 150,
initial_rows = 50,


to my config file..

**Comment by seqizz** [comment]
I'm struggling to make this a bit more dynamic.

```
function set_col_overrides(window, pane)
  local overrides = window:get_config_overrides() or {}
  local pane_obj = pane:get_dimensions()
  overrides.initial_cols = pane_obj.cols
  overrides.initial_rows = pane_obj.viewport_rows
  window:set_config_overrides(overrides)
end

wezterm.on("window-resized", function(window, pane)
  set_col_overrides(window, pane)
end)

wezterm.on("window-config-reloaded", function(window, pane)
  set_col_overrides(window, pane)
end);
```

Which works OK, but I could not make this work on "mux-startup" or "gui-startup" events, even by using exact examples on the documentation. I guess fresh-triggered window is not becoming visible yet so it can't provide the col/row count. Setting it to fullscreen also doesn't work since this is an issue inside the terminal, which is always fullscreen in terms of tiling WMs (see the screenshot below).

Also tried to emit window-resized event as a hacky workaround, right after window from "wezterm connect <myunixdomain>" command becomes visible 😅

Or I'd appreciate if there is easier way to directly set the initial_cols/rows from initial window/pane's dimensions on unexpected resizing via wm on start.

![image](https://user-images.githubusercontent.com/307899/182476949-dc753e21-6113-472f-b69d-6721c73178c4.png)

**Reply by seqizz** [reply]
I also wonder why this doesn't happen at all, when I use "wezterm start" without mux involvement.

**Reply by wez** [reply]
@seqizz Can you open an issue for that? I think the real problem is that the terminal doesn't match the window size in your tiling WM?

**Reply by seqizz** [reply]
Yep, created [this one](https://github.com/wez/wezterm/issues/2351), thanks!

**Comment by washere** [comment]
The answer all over the web & above here & in official reference did not work for me. This DID NOT WORK:

```
initial_rows = 48
initial_cols = 150
-- WRONG ANSWER
```

Needed to add config. to start of both lines.
I found the answer after an hour thanks to Daniel Lizik:

```
-- SIZE of window:
-- difficult to find as manual ref/google-results answer initial_cols = & initial_rows = DO NOT WORK,
-- but config.initial_cols = WORKS! from answer by Daniel-Lizik:
-- https://stackoverflow.com/questions/78738575/how-to-maximize-wezterm-on-startup

config.enable_tab_bar = true
config.initial_rows = 48
config.initial_cols = 150

--config.window_decorations = "NONE"
config.window_decorations = "RESIZE"


```

Hope helps others.

**Reply by ealap** [reply]
From someone who just started using Wezterm today and is currently scouring the forum for answers, I have observed that there are at least two ways of writing the config that are usually given here.

One is building the config and storing it in a variable called `config`.
```lua
local wezterm = require 'wezterm'
local config = wezterm.config_builder()

config.initial_rows = 48
config.initial_cols = 150

return config
```
You could even see some examples that use a different variable name instead of `config`.


Another way is directly returning a config without storing it in a variable.
```lua
return {
    initial_rows = 48,
    initial_cols = 150
}
```
This one is likely what the first answer was going for.
